llvm/test/tools/llvm-objcopy/ELF/change-section-lma-multiple-load.test

```diff
@@ -0,0 +1,52 @@
+# RUN: yaml2obj %s -o %t
```

Comment:
I think you additionally need:
1) A segment with 0 file size.
2) A segment with an explicit p_paddr that is different to the p_vaddr.
3) A non PT_LOAD segment (with non-zero p_filesz), to show that it's not just PT_LOAD segments that are impacted (should they be impacted though? what does GNU objcopy do?).
4) A nested segment (with non-zero p_filesz) to show that segments within other segments are appropriately updated. The segment should probably have a non-zero offset within the PT_LOAD it's nested inside.
5) A segment with 0 p_vaddr/p_paddr.
6) Test cases that cover all the different new error messages you've added for parsing the option.

Reply:
In change-section-lma.test: 
1. PT_NOTE
2. PT_LOAD (.text2) 
3. PT_PHDR
4. PT_PHDR in PR_LOAD (.text1) 
5. PT_NOTE
6. Covers all of them but the use with unsupported file formats which seems excessive

Reply:
> The segment should probably have a non-zero offset within the PT_LOAD it's nested inside.
I am not sure how that helps with coverage.

Reply:
I honestly can't remember what my reasoning was with that case 🤷‍♂️

Reply:
With point 3, I was explicitly referring to a segment not nested within another segment. This is important, because some tools treat PT_LOAD segments (and segments nested within them) as different to other top-level segment types, hence why I'd ideally like this checked against GNU's behaviour.

Reply:
I see. For point 3, I will add another non-zero-size NOTE header in fixup 4 that is independent from the LOAD headers.
As mentioned before in other places I don't see a meaning in GNU's objcopy actual behaviour with this particular option. However, I defined what the LLVM version will do based on GNU documentation, some bits and pieces of GNU behaviour and how I saw the option being used in real software. 

The zero-size NOTE is already independent as it automatically gets virtual address of 0x0.

Reply:
The non-empty PT_NOTE is currently nested inside the first PT_LOAD segment, if I'm not mistaken? I think it needs to have a companion section that can be assigned to it, to force the offset to be not within the other PT_LOADs.

> The zero-size NOTE is already independent as it automatically gets virtual address of 0x0.

Not quite what I meant (I wanted a segment with 0 virtual address, and a different segment with zero file size). That being said, I actually don't think we need a 0 virtual address case, having given it a bit more thought, since there's nothing special about this case.

Reply:
Ah OK, I've changed the test so that the non-zero PT_NOTE section is elsewhere in the file. And then I thought it was better to make the 0-size segment independent too (it might not really matter considering it is zero size).